C-Plus-Plus/cp/Leftmostnonrepeatingchar.cpp

```diff
@@ -0,0 +1,37 @@
+// Description: This program finds the leftmost non repeating character in a string.
+// For example: s = "aabcdb", output should be: 3 (index of c)
+#include<bits/stdc++.h>
+using namespace std;
+
+int leftmostnonrepeatingchar(string s){
+    int res = numeric_limits<int>::max();  //Initialised res to infinity
+    int fi[256];                           //An array to store first indices of string characters(size is 256 because there are 256 ascii characters)
+    for(int i = 0; i < 256; i++){          //Initialising all elements to -1
+        fi[i] = -1;
```

Comment:
Your code works fine but 
I need you to remove this for loop to initialize the fi array to -1
and modify code accordingly.

Reply:
I tried using fi[256] = {-1} but this wasn't initialising all elements to -1

Reply:
initially, the array is default initialized with 0 try to use that

Reply:
But then there will be some issues with 0th index, if we keep all elements of the array are 0, so how do we prevent that